src/XO.java

```diff
@@ -0,0 +1,162 @@
+import java.util.Scanner;
+
+public class XO implements IBoardGame {
+
+    String[][] board;
+    String winner;
+    public XO() {
+        board = new String[3][3];
+
+        winner = null;
+        for(int i=0;i<3;i++)
+        {
+            for(int j=0;j<3;j++)
+            {
+                board[i][j] = "E";
+            }
+        }
+    }
+
+    @Override
+    public String getWinner() {
+
+        return winner;
+    }
+
+    @Override
+    public boolean takeShot(boolean isX, int row, int col) {
+        row--;
+        col--;
+        if(board[row][col]=="E")
+        {
+            if(isX)
+            {
+                board[row][col] = "X";
+                return true;
+            }
+            else
+            {
+                board[row][col] = "O";
+                return true;
+            }
+        }
+        else
+            return false;
+    }
+    public void  getUI()
```

Comment:
ให้แยกการแสดงผลออกจากการคำนวณครับ เพราะไม่อย่างนั้นเวลาที่เราจะเอาเกมนี้ไปเล่นบน เว็บ, มือถือ หรืออะไรก็ตามแต่ เราจะต้องมาแก้โค้ดการคำนวณทุกครั้ง ซึ่งผิดหลักของ Separation of concern ครับ